thicket/tests/test_caliperreader.py

```diff
@@ -23,3 +23,66 @@ def test_from_caliperreader(rajaperf_seq_O3_1M_cali):
         ].values[0]
         == 103.47638
     )
+
+
+def test_node_ordering_from_caliper(caliper_ordered):
+    """Check the order of output from the native Caliper reader by examining a known input with node order column."""
+
+    tk = Thicket.from_caliperreader(caliper_ordered)
+
+    expected_order = [
+        "main",
+        "lulesh.cycle",
+        "TimeIncrement",
+        "LagrangeLeapFrog",
+        "LagrangeNodal",
+        "CalcForceForNodes",
+        "CalcVolumeForceForElems",
+        "IntegrateStressForElems",
+        "CalcHourglassControlForElems",
+        "CalcFBHourglassForceForElems",
+        "LagrangeElements",
+        "CalcLagrangeElements",
+        "CalcKinematicsForElems",
+        "CalcQForElems",
+        "CalcMonotonicQForElems",
+        "ApplyMaterialPropertiesForElems",
+        "EvalEOSForElems",
+        "CalcEnergyForElems",
+        "CalcTimeConstraintsForElems",
+    ]
+    expected_data_order = [
+        1.250952,
+        1.229935,
+        0.000085,
+        1.229702,
+        0.604766,
+        0.566399,
+        0.561237,
+        0.161196,
+        0.395344,
+        0.239849,
+        0.614079,
+        0.175102,
+        0.168127,
+        0.136318,
+        0.038575,
+        0.299062,
+        0.293046,
+        0.190395,
+        0.010707,
+    ]
+
+    # check if the rows are in the expected order
+    for i in range(0, tk.dataframe.shape[0]):
+        node_name = tk.dataframe.iloc[i]["name"]
+        assert node_name == expected_order[i]
+        node_data = tk.dataframe.iloc[i]["Total time"]
+        assert node_data == expected_data_order[i]
+
+    # check the tree ordering is correct as well
+    output = tk.tree(metric_column="nid")
+    for i in range(1, 20):
+        location = output.find(str(i))
+        assert location != 0
```

Comment:
```suggestion
    tk.dataframe["hnid"] = [node._hatchet_nid for node in tk.graph.node_order_traverse()]
    output = tk.tree(metric_column="hnid", render_header=False, precision=3)
    for i in tk.dataframe["hnid"].tolist():
        location = output.find(str(i)+".000")
        assert location != -1
```
A couple of things I noticed in this check:

1. The `nid` column is not the same as the `_hatchet_nid`, so to be explicit we should check for the `_hatchet_nid`. In this case the `nid` = `_hatchet_nid + 1`, I'm not sure if there's a case where they would be in different orders.
2. Render the tree with `hnid`, set the header to false, and set the precision to 3 for the find function. 3 is the default but just in case the default changes.
3. check `location != -1` instead of `0` as `find` will return `-1` if the value is not found.

Reply:
Good idea, I think we should be as explicit as possible.